readthedocs/config/models.py

```diff
@@ -2,36 +2,78 @@
 
 from __future__ import division, print_function, unicode_literals
 
-from collections import namedtuple
```

Comment:
So, I had to replace the namedtuples for classes with `__slots__`. Why? Because of https://github.com/rtfd/readthedocs.org/pull/4749 we need to convert the config object to a dict, I was using `_asdict` method from the namedtuples, but isn't recursive, and we can't expand that behavior.

Also, this way is easier to identify their parameters and was able to do the change very smoothly, all tests keep passing.

Reply:
Here is a quick explanation of `__slots__` https://stackoverflow.com/a/28059785/5689214

I didn't put so much attention in the memory reduction but in the way that doesn't allow to put more attributes dynamically, that way we save bugs because of typos.

`__fields__`, `__attrs__` and `__expected_attributes__` don't look like a standard thing in python p:

Reply:
:+1: I wasn't aware that `__slots__` was Python magic. If none of this would be possible by extending namedtuple, I think this implementation works well!